specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/natGateway.json

```diff
@@ -0,0 +1,407 @@
+          },
+          {
+            "name": "natGatewayName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the nat gateway."
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "$expand",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "Expands referenced resources."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns the resulting NatGateway resource.",
+            "schema": {
+              "$ref": "#/definitions/NatGateway"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get nat gateway": { "$ref": "./examples/NatGatewayGet.json" }
+        }
+      },
+      "put": {
+        "tags": [
+          "NatGateways"
+        ],
+        "operationId": "NatGateways_CreateOrUpdate",
+        "description": "Creates or updates a nat gateway.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "natGatewayName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the nat gateway."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/NatGateway"
+            },
+            "description": "Parameters supplied to the create or update nat gateway operation."
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Create successful. The operation returns the resulting NatGateway resource.",
+            "schema": {
+              "$ref": "#/definitions/NatGateway"
+            }
+          },
+          "200": {
+            "description": "Update successful. The operation returns the resulting NatGateway resource.",
+            "schema": {
+              "$ref": "#/definitions/NatGateway"
+            }
+          },
+          "202": {
+            "description": "Accepted and the operation will complete asynchronously."
+          }
+        },
+        "x-ms-examples": {
+          "Create nat gateway": { "$ref": "./examples/NatGatewayCreate.json" }
+        },
+        "x-ms-long-running-operation": true
```

Comment:
Please add `x-ms-long-running-operation-operations`
https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#R2010
For `PUT` it should be `azure-async-operation`

Reply:
thanks, updated